src/aind_behavior_curriculum/tmp.py

```diff
@@ -0,0 +1,80 @@
+"""
+Work in Progress Curriculum
+"""
+
+from collections import defaultdict
+
+class Task:
+    """
+    Defined in exisiting PR
+    """
+    pass
+
+class Metric:
+    """
+    Initalizes a connection to relevant database (referencing some configuration file)
+    User extends this with their own metric implementation based on data inside the database.
+    """
+    pass
+
+
+class Evaluation:
+    """
+    Serializable Callable, takes in a list of metrics that is serialized alongside.
+    """
+    pass
+
+
+# TODO: De/serialization
+class Curriculum:
+    def __init__(self) -> None:
+        TaskId = int
+        # ID tasks on Task.name, a required field.
+        # Tasks are allowed to change parameters.
+        self.tasks: dict[str, TaskId] = {}
```

Comment:
The two collections should not have setters. The collections should not be modifiable using the add/pop methods otherwise it will be too easy to break the graph.